src/wallet/feebumper.cpp

```diff
@@ -265,11 +259,9 @@ Result CommitTransaction(CWallet* wallet, const uint256& txid, CMutableTransacti
     wtxBumped.fTimeReceivedIsTxTime = true;
     wtxBumped.fFromMe = true;
     CValidationState state;
-    if (!wallet->CommitTransaction(wtxBumped, reservekey, g_connman.get(), state)) {
-        // NOTE: CommitTransaction never returns false, so this should never happen.
-        errors.push_back(strprintf("The transaction was rejected: %s", state.GetRejectReason()));
-        return Result::WALLET_ERROR;
-    }
+    bool result = wallet->CommitTransaction(wtxBumped, reservekey, g_connman.get(), state));
```

Comment:
Rather than asserting that the result is true, I think it would make more sense to change the signature of CommitTransaction so that it doesn't return anything.

Reply:
See https://github.com/bitcoin/bitcoin/pull/12361#discussion_r166472325.